### PR TITLE
[ioctl] Build action transfer command line into new ioctl

### LINKED ISSUE #3323: [ioctl] build action transfer command line into new ioctl
### What would you like to be added
 - Step 1: create `actiontransfer.go` and `actiontransfer_test.go` files in `./ioctl/newcmd/action`. create this directory if not exist. 
 - Step 2: refactor  command  https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/action/actiontransfer.go in `actiontransfer.go`
 - Step 3: add unit test in xx_test.go

#### Notes:
1. Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
2. [Output package](https://github.com/iotexproject/iotex-core/blob/master/ioctl/output) is deprecated. `e.g.` `output.PrintError` could be replaced with `errors.Wrap`.
3. Global variables (config.ReadConfig  config.DefaultConfigFile etc.) have been wrapped into client interface. Please use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to access them if needed
4. The logic of new code should be similar to [the old one](https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/action/actiontransfer.go).
5. In each unit test, the results should be checked whether they are as the expected. 

#### Example:
1. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountinfo.go
2. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountdelete.go
3. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountcreateadd.go

### Why is this needed
Unit tests couldn't be added for the old ioctl command (./ioctl/cmd)

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description
Build `actiontransfer` command in new `ioctl`, with the following note.

* Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
* Output package is deprecated, replace it with errors package.
* Replace fmt.Println with cmd.Println
* Refactor unit test to cover the modification.

Fixes #3323 

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [x] TestNewActionTransferCmd

**Test Configuration**:
- Firmware version: Ubuntu 21.04
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules****

### CODE REVIEW

File: ioctl/newcmd/action/action.go [hunk 1]
```diff
@@ -208,8 +208,8 @@ func NewActionCmd(client ioctl.Client) *cobra.Command {
 	}
 
 	// TODO add sub commands
-	// cmd.AddCommand(NewActionHash(client))
-	// cmd.AddCommand(NewActionTransfer(client))
+	ac.AddCommand(NewActionHashCmd(client))
```
Comment (huof6829) [inline ioctl/newcmd/action/action.go:211]: can you fix line389~394 pls?

File: ioctl/newcmd/action/actiontransfer_test.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+		},
+	}
+
+	apiServiceClient.EXPECT().GetAccount(gomock.Any(), gomock.Any()).Return(accountResp, nil).Times(18)
+	apiServiceClient.EXPECT().GetChainMeta(gomock.Any(), gomock.Any()).Return(chainMetaResp, nil).Times(6)
+	apiServiceClient.EXPECT().SendAction(gomock.Any(), gomock.Any()).Return(sendActionResp, nil).Times(5)
+
+	t.Run("action transfer", func(t *testing.T) {
+		cmd := NewActionTransferCmd(client)
+		RegisterWriteCommand(client, cmd)
```
Comment (huof6829) [inline ioctl/newcmd/action/actiontransfer_test.go:70]: add in `NewActionCmd`

File: ioctl/newcmd/action/actiontransfer.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+			}
+			gasPriceRau, err := gasPriceInRau(client, cmd)
+			if err != nil {
+				return errors.Wrap(err, "failed to get gas price")
+			}
+			nonce, err := nonce(client, cmd, sender)
+			if err != nil {
+				return errors.Wrap(err, "failed to get nonce")
+			}
+			tx, _ := action.NewTransfer(nonce, amount,
```
Comment (huof6829) [inline ioctl/newcmd/action/actiontransfer.go:87]: check error

Comment (LuckyPigeon) [reply ioctl/newcmd/action/actiontransfer.go:87]: `action.NewTransfer` never returns error, do we still need to check?

Comment (huof6829) [reply ioctl/newcmd/action/actiontransfer.go:87]: it's better to check error

File: ioctl/newcmd/action/actiontransfer.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+			"价格] [-P 密码] [-y]",
+	}
+)
+
+// NewActionTransferCmd represents the action transfer command
+func NewActionTransferCmd(client ioctl.Client) *cobra.Command {
+	use, _ := client.SelectTranslation(_actionTransferCmdUses)
+	short, _ := client.SelectTranslation(_actionTransferCmdShorts)
+
+	cmd := &cobra.Command{
```
Comment (huof6829) [inline ioctl/newcmd/action/actiontransfer.go:40]: directly return

File: ioctl/newcmd/action/action.go [hunk 5]
```diff
@@ -208,15 +208,17 @@ func NewActionCmd(client ioctl.Client) *cobra.Command {
...
+	ac.AddCommand(NewActionTransferCmd(client))
 	// cmd.AddCommand(NewActionDeploy(client))
 	// cmd.AddCommand(NewActionInvoke(client))
 	// cmd.AddCommand(NewActionRead(client))
 	// cmd.AddCommand(NewActionClaim(client))
 	// cmd.AddCommand(NewActionDeposit(client))
 	// cmd.AddCommand(NewActionSendRaw(client))
 
+	RegisterWriteCommand(client, ac)
```
Comment (huof6829) [inline ioctl/newcmd/action/action.go:220]: the second args should be  actionTransferCmd

**Review by huof6829 — APPROVED** [review]

**Comment by LuckyPigeon** [comment]
@huof6829 Thanks for the improvement!

File: ioctl/newcmd/action/action.go [hunk 6]
```diff
@@ -146,8 +146,8 @@ func NewActionCmd(client ioctl.Client) *cobra.Command {
 	}
 
 	// TODO add sub commands
-	// cmd.AddCommand(NewActionHash(client))
-	// cmd.AddCommand(NewActionTransfer(client))
+	ac.AddCommand(NewActionHashCmd(client))
```
Comment (Liuhaai) [inline ioctl/newcmd/action/action.go:149]: `ac`?

**Comment by Liuhaai** [comment]
@LuckyPigeon could you help the failure of test?

File: ioctl/newcmd/action/action.go (hunk 6 above)
Comment (huof6829) [reply ioctl/newcmd/action/action.go]: fixed

**Comment by huof6829** [comment]
> @LuckyPigeon could you help the failure of test?

fixed

**Comment by LuckyPigeon** [comment]
@huof6829 Thanks for the help!

**Review by Liuhaai — APPROVED** [review]

**Pull request merged**